src/main.rs

```diff
@@ -0,0 +1,200 @@
+                    None => {
+                        return Ok(HttpResponse::new(http::StatusCode::UNAUTHORIZED));
+                    },
+                })).json(&map).send() {
+                    Ok(res) => res,
+                    Err(err) => {
+                        error!("{}", err);
+                        return Ok(HttpResponse::new(http::StatusCode::INTERNAL_SERVER_ERROR))
+                    }
+                };
+            info!("{:?}", response);
+            Ok(HttpResponse::new(http::StatusCode::from_u16(response.status().as_u16()).unwrap()))
+        }).responder()
+}
+
+fn healthcheck(_: HttpRequest) -> &'static str {
+    "✅"
+}
+
+fn main() {
+    setup_logging(log::LevelFilter::Debug);
+    match server::new(|| {
+        App::new()
+            .middleware(Logger::default())
+            .route("/", http::Method::POST, hook)
+            .route("/health", http::Method::GET, healthcheck)
+    }).bind("0.0.0.0:8888")
+    {
+        Ok(server) => server.run(),
+        Err(err) => error!("Couldn't start server: {}", err),
+    }
+}
+
+fn setup_logging(level: log::LevelFilter) {
+    match fern::Dispatch::new()
+        .format(|out, message, record| {
+            out.finish(format_args!(
+                "[{}][{}] {}",
+                chrono::Local::now().format("%Y-%m-%d %H:%M:%S"),
+                record.level(),
+                message
+            ))
+        })
+        .level(level)
+        .chain(std::io::stdout())
+        .apply()
+    {
+        Err(_) => {
+            eprintln!("error setting up logging!");
+        }
+        _ => info!("logging set up properly"),
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AlertHook {
+    #[serde(rename = "version")]
+    pub(crate) version: String,
+
+    #[serde(rename = "groupKey")]
+    pub(crate) group_key: String,
+
+    #[serde(rename = "status")]
+    pub(crate) status: Status,
+
+    #[serde(rename = "receiver")]
+    pub(crate) receiver: String,
+
+    #[serde(rename = "groupLabels")]
+    pub(crate) group_labels: HashMap<String, String>,
+
+    #[serde(rename = "commonLabels")]
+    pub(crate) common_labels: HashMap<String, String>,
+
+    #[serde(rename = "commonAnnotations")]
+    pub(crate) common_annotations: HashMap<String, String>,
+
+    #[serde(rename = "externalURL")]
+    pub(crate) external_url: String,
+
+    #[serde(rename = "alerts")]
+    pub(crate) alerts: Vec<Alert>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Alert {
+    #[serde(rename = "labels")]
+    pub(crate) labels: HashMap<String, String>,
+
+    #[serde(rename = "annotations")]
+    pub(crate) annotations: CachetAnnotation,
+
+    #[serde(rename = "startsAt")]
+    pub(crate) starts_at: String,
+
+    #[serde(rename = "endsAt")]
+    pub(crate) ends_at: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Status {
+    #[serde(rename = "firing")]
+    Firing,
+
+    #[serde(rename = "resolved")]
+    Resolved,
+}
+
```

Comment:
A comment here on why we're doing this extra work would be good for future selves.

Reply:
Did comment a bit further down